File: app/scala/src/main/scala/com/example/accounting/domain/journal/JournalEntry.scala
package com.example.accounting.domain.journal

import java.time.{LocalDate, LocalDateTime}

/**
 * 仕訳エントリ（2層構造のヘッダー）
 */
case class JournalEntry(
    voucherNumber: String,
    journalDate: LocalDate,
    description: String,
    totalAmount: BigDecimal,
    referenceNumber: Option[String],
    creator: String,
    updater: Option[String],
    createdAt: LocalDateTime,
    updatedAt: LocalDateTime,
)

/**
 * 仕訳明細（2層構造の明細）
 */
case class JournalEntryDetail(
    voucherNumber: String,
    lineNumber: Int,
    accountCode: String,
    debitAmount: BigDecimal,
    creditAmount: BigDecimal,
    description: String,
    taxAmount: Option[BigDecimal],
    taxRate: Option[BigDecimal],
)

File: app/ruby/accounting_service/financial_accounting/app/domain/models/journal_entry.rb
# frozen_string_literal: true

module Domain
  module Models
    class JournalEntry
      attr_accessor :journal_id, :line_number, :account_code, :debit_amount, :credit_amount, :description

      def initialize(account_code:, debit_amount: 0, credit_amount: 0, description: '')
        @account_code = account_code
        @debit_amount = BigDecimal(debit_amount.to_s)
        @credit_amount = BigDecimal(credit_amount.to_s)
        @description = description
      end
    end
  end
end
